Ekubo wETH-ETH liquidity pool.ipynb

```diff
@@ -0,0 +1,1405 @@
+    "        'price': 1.000001**(-combined_events_response_swap[event_no_swap].data[18] * bool_to_sign(combined_events_response_swap[event_no].data[19])),\n",
+    "        'tick_id': -combined_events_response_swap[event_no_swap].data[18] * bool_to_sign(combined_events_response_swap[event_no].data[19]),\n",
+    "        'liquidity_after': combined_events_response_swap[event_no_swap].data[20],\n",
+    "        'amount0': combined_events_response_swap[event_no_swap].data[12] * bool_to_sign(combined_events_response_swap[event_no].data[13]), # 0 swapping +ve \n",
+    "        'amount1': combined_events_response_swap[event_no_swap].data[14] * bool_to_sign(combined_events_response_swap[event_no].data[15])\n",
+    "    }\n",
+    "    if (hex(combined_events_response_swap[event_no_swap].data[2]) == \"0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7\" and \\\n",
+    "        hex(combined_events_response_swap[event_no_swap].data[1]) == \"0x42b8f0484674ca266ac5d08e4ac6a3fe65bd3129795def2dca5c34ecc5f96d2\" and \\\n",
+    "        combined_events_response_swap[event_no_swap].data[3] == pool_fee_touse and \\\n",
+    "        combined_events_response_swap[event_no_swap].data[4] == tick_spacing_touse):\n",
+    "            ekubo_eth_weth_swap_data.append(event_data)\n",
+    "    \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "4e1024a0",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "123100\n",
+      "158440\n",
+      "162969\n",
+      "158443\n",
+      "0.01\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Create a pandas DataFrame\n",
+    "liquidity_df = pd.DataFrame(ekubo_eth_weth_liquidity_data)\n",
+    "print(min(liquidity_df.tick_lower))\n",
+    "print(min(liquidity_df.tick_upper))\n",
+    "\n",
+    "swap_df = pd.DataFrame(ekubo_eth_weth_swap_data)\n",
+    "print(max(swap_df.tick_id))\n",
+    "print(min(swap_df.tick_id))\n",
+    "\n",
+    "\n",
+    "swap_df['fee_token1_ETH'] = (pool_fee/100 * swap_df['amount1'].abs()) / 10**18\n",
+    "\n",
+    "print(max(swap_df.fee_token1_ETH))\n",
+    "\n",
+    "# # Convert to Parquet using pyarrow\n",
+    "# table = pa.Table.from_pandas(df)\n",
+    "# pq.write_table(table, 'events_data.parquet')\n",
+    "\n",
+    "# print(\"Data saved to Parquet file successfully!\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "d513668c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Step 1: Collect block numbers and prices from ekubo_eth_weth_swap_data\n",
+    "block_price_data = [{'block_number': event['block_number'], 'price': event['price']} for event in ekubo_eth_weth_swap_data]\n",
+    "\n",
+    "# Step 2: Convert the collected data into a pandas DataFrame\n",
+    "df = pd.DataFrame(block_price_data)\n",
+    "\n",
+    "# Step 3: Group by block number and calculate the median price for each block\n",
```

Comment:
You could have just done on ticks directly (i.e. median tick value, instead of price) bcz finally for storing liquidity, u need tick range.